activity-stream-impl/src/main/marathon-resources/platform.conf

```diff
@@ -0,0 +1,26 @@
+akka {
+  io {
+    dns {
+      resolver = async-dns
+      async-dns {
+        resolve-srv = true
+        resolv-conf = on
+      }
+    }
+  }
+}
+
+service-locator-dns {
+  name-translators = [
+    { "^_.+$"               = "$0" },
+    {  "^activityservice$"  = "_activity-stream-impl-chirper._tcp.marathon.mesos" },
+    {  "^chirpservice$"     = "_chirp-impl-chirper._tcp.marathon.mesos" },
+    {  "^friendservice$"    = "_friend-impl-chirper._tcp.marathon.mesos" },
```

Comment:
It's a bit unfortunate that each dependency has to be listed explicitly. How are the marathon service names determined? Is there a way we could make it use something based on the service descriptor name instead of the implementation project name?

Reply:
It's determined by the `id` field for each app e.g. `id` of `/chirper/activity-stream-impl` becomes an SRV entry of `_activity-stream-impl-chirper._tcp.marathon.mesos`.

I listed these out because we're going to have our tooling do that (since it has static knowledge of all of the services), so this example would then be consistent with the tooling.

On the other hand, we probably could make this work more like the Kubernetes example since this is indeed handcrafted and kind of a pain to maintain.

We'd have to rename the apps in the `marathon.json` to be named after their service name (so `id` of `/chirper/activityservice`) and then these entries could be collapsed into one.

I'd imagine it would be something like `"^(.+)$" = "_$1-chirper._tcp.marathon.mesos"`

WDYT?

Reply:
I think the existing approach will break down for systems that don't put all services into the same build/git repo (see [Splitting a system into multiple builds](https://www.lagomframework.com/documentation/1.3.x/scala/MultipleBuilds.html)).

The name `activity-stream-impl` is an implementation detail and not part of the public contract of the service. So, yes, I think it would be better to use the service name as listed in the descriptor (i.e., "activityservice" in this case) to generate the `id` field.

I'm not sure about the "/chirper" prefix/"-chirper" suffix. In the tooling, where does it get that component from? Maybe it would be simpler to use a flat list of services?

Reply:
Or does the tooling has some notion of a namespace that maps to k8s namespaces and Marathon application groups?

Reply:
Thanks for that - I think flattening may make sense from the Marathon side of things. We haven't implemented anything there so we're just trying to track how it _could_ work if we do it some day. Currently there isn't any namespace concept in the tooling but I could see that being a pretty trivial addition.

Thanks for bringing up the consideration of multiple builds. Indeed, listing services won't work then and we'll need a different approach. One thought: is there a limitation/validation for the service name from the Lagom side?

For instance, let's say you couldn't have a period (`.`) in your service name.

Mapping `^[^.]+$` to `_http-lagom-api._tcp.$0.default.svc.cluster.local` would work pretty well I think. This would automatically exclude any commonly used DNS names as you usually have at least one domain.

Reply:
Lagom doesn't currently impose any restrictions on the service name, no, but maybe it's a good idea (or at least to point out that the deployment environment might impose some restrictions).

I was just looking at how DC/OS handles [service naming](https://docs.mesosphere.com/1.10/networking/mesos-dns/service-naming/) and there's a lot to it:

> Mesos-DNS follows RFC 952 for name formatting. All fields used to construct hostnames for A records and service names for SRV records must be 24 characters or shorter and can include letters of the alphabet (A-Z), numbers (0-9), and a dash (-). Names are not case sensitive. If the task name does not comply with these constraints, Mesos-DNS will shorten the name to 24 characters, remove all invalid characters, and replace periods (.) with a dash (-).

It's probably smart for us to use the same restrictions. Raised as https://github.com/lagom/lagom/issues/1085.